src/events.rs

```diff
@@ -72,19 +73,32 @@ impl std::fmt::Debug for RawEvent {
 }
 
 /// Events decoder.
-#[derive(Debug)]
 pub struct EventsDecoder<T> {
     metadata: Metadata,
     type_sizes: HashMap<String, usize>,
+    type_segmenters: HashMap<
+        String,
+        Box<dyn Fn(&mut &[u8], &mut Vec<u8>) -> Result<(), Error> + Send>,
```

Comment:
I'd prefer introducing a trait here for readability

Reply:
Done